hector_pointcloud_decimator/src/pointcloud_decimator.cpp

```diff
@@ -0,0 +1,174 @@
+  // point_fraction:         Fraction of points to keep
+  // point_count:            Number of points to keep
+
+  declare_reconfigurable_parameter( "input", std::ref( input_ ), "The input topic for pointclouds",
+                                    hector::ParameterOptions<std::string>().onUpdate(
+                                        [this]( const std::string & ) { this->setup(); } ) );
+  // Base the default output topic on the chosen input topic
+  output_ = input_ + "_decimated";
+  declare_reconfigurable_parameter(
+      "output", std::ref( output_ ), "The output topic for pointcloud_transport",
+      hector::ParameterOptions<std::string>().onUpdate( [this]( const auto & ) { this->setup(); } ) );
+  declare_reconfigurable_parameter(
+      "elimination_method", std::ref( elimination_method_ ),
+      "The method used to decimate the pointcloud",
+      hector::ParameterOptions<std::string>()
+          .setAdditionalConstraints( "Allowed values: 'random', 'count'" )
+          .onValidate(
+              []( const std::string &value ) { return value == "random" || value == "count"; } ) );
+  declare_reconfigurable_parameter(
+      "elimination_quantifier", std::ref( elimination_quantifier_ ),
+      "How the amount of points to be eliminated is specified",
+      hector::ParameterOptions<std::string>()
+          .setAdditionalConstraints( "Allowed values: 'fraction', 'count'" )
+          .onValidate( []( const std::string &value ) {
+            return value == "fraction" || value == "count";
+          } ) );
+  declare_reconfigurable_parameter(
+      "point_fraction", std::ref( point_fraction_ ), "The fraction of points to keep",
+      hector::ParameterOptions<double>()
+          .setAdditionalConstraints( "Mutually exclusive with point_total" )
+          .onValidate( []( const double &value ) { return value >= 0.0 && value <= 1.0; } ) );
+  declare_reconfigurable_parameter(
+      "point_count", std::ref( point_count_ ), "The total number of points to keep",
+      hector::ParameterOptions<int>().onValidate( []( const int &value ) { return value >= 0; } ) );
+
+  pct_node_ = std::make_shared<rclcpp::Node>( static_cast<std::string>( get_name() ) + "_pct" );
+  pct_ = std::make_unique<point_cloud_transport::PointCloudTransport>( pct_node_ );
+  setup();
+}
+
+void PointcloudDecimator::setup()
+{
+  // subscriber for handling incoming messages
+  pointcloud_subscriber_ = create_subscription<sensor_msgs::msg::PointCloud2>(
+      input_, 10, std::bind( &PointcloudDecimator::pointcloudCallback, this, std::placeholders::_1 ) );
+  RCLCPP_INFO( get_logger(), "Subscribed to '%s'", pointcloud_subscriber_->get_topic_name() );
+
+  // publisher for publishing outgoing messages
+  pointcloud_publisher_ = pct_->advertise( output_, 10 );
+  RCLCPP_INFO( get_logger(), "Publishing results to '%s'", pointcloud_publisher_.getTopic().c_str() );
+
+  check_subscribers_timer_ =
+      create_wall_timer( std::chrono::seconds( 1 ),
```

Comment:
I would check this more often, as this check is not expensive. Maybe every 100ms.